src/tap/cmd.rs

```diff
@@ -70,20 +80,30 @@ pub async fn cmd(opts: &super::Opts, mut signal_rx: SignalRx) -> exitcode::ExitC
         let stream = res.stream();
     };
 
-    // Loop over the returned results, printing out log events.
-    // NOTE: This will currently ignore notifications. A later `--verbose` option is planned
-    // to include these.
-    // TODO: https://github.com/vectordotdev/vector/issues/6870
+    // Loop over the returned results, printing out tap events.
     loop {
         tokio::select! {
             biased;
             Some(SignalTo::Shutdown | SignalTo::Quit) = signal_rx.recv() => break,
             Some(Some(res)) = stream.next() => {
                 if let Some(d) = res.data {
-                    for event_string in d.output_events_by_component_id_patterns.iter().filter_map(|ev| ev.as_string()) {
-                        #[allow(clippy::print_stdout)]
-                        {
-                            println!("{}", event_string);
+                    for tap_event in d.output_events_by_component_id_patterns.iter() {
+                        match tap_event {
+                            OutputEventsByComponentIdPatternsSubscriptionOutputEventsByComponentIdPatterns::Log(ev) => {
+                                println!("{}", ev.string);
+                            },
+                            OutputEventsByComponentIdPatternsSubscriptionOutputEventsByComponentIdPatterns::Metric(ev) => {
+                                println!("{}", ev.string);
+                            },
+                            OutputEventsByComponentIdPatternsSubscriptionOutputEventsByComponentIdPatterns::EventNotification(ev) => {
+                                if opts.verbose {
+                                    match ev.notification {
+                                        EventNotificationType::MATCHED => println!(r#"[tap] Pattern "{}" successfully matched."#, ev.pattern),
```

Comment:
Can we print this to stderr too? I'd like to preserve stdout of `vector tap` as _just_ the tapped events so that users can easily pipe or redirect the output.

Reply:
👍🏻 I think this is especially important given that error messages are guaranteed to break the serialization format of events.

Reply:
Cool, added in [6ad468f](https://github.com/vectordotdev/vector/pull/11321/commits/6ad468f319907e762784855c14dc650a7ea8e4df)